extensions/mattermost/src/channels.tsx

```diff
@@ -265,30 +255,32 @@ function ChannelList(props: { profile: UserProfile; team: TeamUI }) {
       })();
     });
 
-    setLoading(false);
     showToast(Toast.Style.Success, `${team.name} channels`);
 
     return categoriesUI;
   }
 
-  useEffect(() => {
-    (async () => {
-      team.categories = await loadCategories();
-      setTeam(team);
-    })();
-  }, []);
+  const { isLoading } = usePromise(async () => {
+    team.categories = await loadCategories();
+    setTeam(team);
+  });
 
   async function openChannel(channel: ChannelUI) {
     console.log("select channel", channel);
     const baseDeeplink = preference.baseUrl.replace("https://", "mattermost://");
     const fullDeeplink = baseDeeplink + "/" + team.name + channel.path;
     console.log("open deeplink", fullDeeplink);
-    open(fullDeeplink);
-    closeMainWindow();
+    try {
+      await open(fullDeeplink);
```

Comment:
Are you sure this is working correctly? After the application has opened, it takes some time for the launch screen to disappear. If you try to open a deep link earlier, this deep link will most likely be ignored. (I remember encountering a similar problem. I will try to check this behavior again)

Reply:
I'm not using the Mattermost application so I don't really know 🙁.
I simply enclosed the code in a try...catch otherwise it is the same as you wrote it originally. One thing we could do is check if the application is running first e.g. through `run-applescript` and if app is running, THEN open the deeplink otherwise show a failure toast asking user to launch Mattermost application. Does that make sense?

Reply:
OK check out the latest commit: 
we have an AppleScript function that checks if Mattermost is running first.
Hopefully that helps.

Reply:
Oops, sorry. I didn't notice that the logic for opening the app under the launchMattermostAppOnLoad flag was still there.
In that case, the current behavior shouldn't break.

----

However, instead of the preference flag launchMattermostAppOnLoad, it is worthwhile to automatically determine whether the application is selected as a method for opening deep links.
Examle of new preference in package.json
```json
 {
      "name": "application",
      "type": "dropdown",
      "required": false,
      "title": "Open In",
      "data": [
        {
          "title": "Mattermost Application",
          "value": "mattermost"
        },
        {
          "title": "Browser",
          "value": "browser"
        }
      ],
      "description": "How to open a deep link",
      "default": "mattermost"
    }
```

And then add expression:
```
if (preference.application === "mattermost") {
   runAppleScriptSync('tell application "Mattermost" to launch');
}
```

And compute url  to open:
```ts
const baseDeeplink = (() => {
      switch (preference.application) {
        case "browser": return preference.baseUrl
        case "mattermost": return preference.baseUrl.replace("https://", "mattermost://");
      }
    })()
```

Reply:
Love this idea.
Yeah let's go with this it makes a lot of sense!

Reply:
I used slightly different wording to what you suggested:
`deepLinkType` for the "name" as it's more obvious than "application"
`application` instead of "mattermost" for application option as that is also more precise.